lib/from_openstudio/construction/shade.rb

```diff
@@ -0,0 +1,72 @@
+#
+# (2) Redistributions in binary form must reproduce the above copyright notice,
+# this list of conditions and the following disclaimer in the documentation
+# and/or other materials provided with the distribution.
+#
+# (3) Neither the name of the copyright holder nor the names of any contributors
+# may be used to endorse or promote products derived from this software without
+# specific prior written permission from the respective party.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDER(S) AND ANY CONTRIBUTORS
+# "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO,
+# THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+# ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER(S), ANY CONTRIBUTORS, THE
+# UNITED STATES GOVERNMENT, OR THE UNITED STATES DEPARTMENT OF ENERGY, NOR ANY OF
+# THEIR EMPLOYEES, BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+# EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT
+# OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+# INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT,
+# STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY
+# OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+# *******************************************************************************
+
+require 'honeybee/construction/shade'
+require 'from_openstudio/model_object'
+
+module Honeybee
+  class ShadeConstruction < ModelObject
+
+    def self.from_construction(construction)
+        # create an empty hash
+        hash = {}
+        hash[:type] = 'ShadeConstruction'
+        # set hash values from OpenStudio Object
+        hash[:identifier] = construction.nameString
+        # get outermost construction layers
+        layer = construction.layers[0]
+        if layer.to_StandardGlazing.is_initialized
+          hash[:is_specular] = true
+          # set reflectance properties from outermost layer
+          #TODO: these properties are giving an `undefined method` OS error
+          #unless layer.frontSideSolarReflectanceatNormalIncidence.empty?
+          #  hash[:solar_reflectance] = layer.frontSideSolarReflectanceatNormalIncidence.get
+          #end
+          #unless layer.frontSideVisibleReflectanceatNormalIncidence.empty?
+          #  hash[:visible_reflectance] = layer.frontSideVisibleReflectanceatNormalIncidence.get
+          #end
+        elsif layer.to_StandardOpaqueMaterial.is_initialized
+          hash[:is_specular] = false
+          # set reflectance properties from outermost layer
+          #TODO: these properties are giving an `undefined method` OS error
+          #unless layer.solarReflectance.empty?
+          #  hash[:solar_reflectance] = layer.solarReflectance.get
+          #end
+          #unless layer.visibleReflectance.empty?
+          #  hash[:visible_reflectance] = layer.visibleReflectance
```

Comment:
All of these commented out properties are throwing an Openstudio `undefined method` error. These methods do exist for the respective material types as boost optionals.


https://openstudio-sdk-documentation.s3.amazonaws.com/cpp/OpenStudio-3.1.0-doc/model/html/classopenstudio_1_1model_1_1_standard_glazing.html

https://openstudio-sdk-documentation.s3.amazonaws.com/cpp/OpenStudio-3.1.0-doc/model/html/classopenstudio_1_1model_1_1_standard_opaque_material.html